moveit_core/robot_trajectory/src/robot_trajectory.cpp

```diff
@@ -347,24 +353,25 @@ void RobotTrajectory::setRobotTrajectoryMsg(const robot_state::RobotState& refer
       st->setVariableAccelerations(trajectory.joint_names, trajectory.points[i].accelerations);
     if (!trajectory.points[i].effort.empty())
       st->setVariableEffort(trajectory.joint_names, trajectory.points[i].effort);
-    addSuffixWayPoint(st, (this_time_stamp - last_time_stamp).toSec());
+    addSuffixWayPoint(st, (this_time_stamp.seconds() - last_time_stamp.seconds()));
```

Comment:
Maybe a minor point here but you can just subtract these two and then call `.seconds()` on the result to keep precision.

From the seconds documentation:

>   /// \return the duration in seconds as a floating point number.
>   /// \warning Depending on sizeof(double) there could be significant precision loss.
>   /// When an exact time is required use nanoseconds() instead

Reply:
e6e7a0b645d756c2c406a526cc51ae4fbbf4bd3d